Zend/zend_ast.c

```diff
@@ -776,12 +776,22 @@ ZEND_API zend_result ZEND_FASTCALL zend_ast_evaluate(zval *result, zend_ast *ast
 			zend_string *case_name = zend_ast_get_str(case_name_ast);
 
 			zend_ast *case_value_ast = ast->child[2];
-			zval *case_value_zv = case_value_ast != NULL
-				? zend_ast_get_zval(case_value_ast)
-				: NULL;
+
+			zval case_value_zv;
+			zval *case_value_zv_ptr = NULL;
+			if (case_value_ast != NULL) {
+				if (UNEXPECTED(zend_ast_evaluate(&case_value_zv, case_value_ast, scope) != SUCCESS)) {
+					ret = FAILURE;
+					break;
+				}
+				case_value_zv_ptr = &case_value_zv;
+			}
 
 			zend_class_entry *ce = zend_lookup_class(class_name);
-			zend_enum_new(result, ce, case_name, case_value_zv);
+			zend_enum_new(result, ce, case_name, case_value_zv_ptr);
+			if (case_value_zv_ptr != NULL) {
+				Z_TRY_DELREF_P(case_value_zv_ptr);
```

Comment:
`zval_ptr_dtor_nogc`, Z_TRY_DELREF_P is not for general use.

Reply:
I still get confused about which ones are to use because there are so many. I'll check better next time.